app/services/courses/generate_course_title_service.rb

```diff
@@ -0,0 +1,54 @@
+module Courses
+  class GenerateCourseTitleService
+    def execute(course:)
+      subjects = course.subjects
+
+      title = if course.further_education_course?
+                further_education_title
+              elsif is_modern_lanuage_course?(subjects)
+                modern_language_title(subjects)
+              else
+                generated_title(subjects)
+              end
+
+      title = append_send_info(title) if course.is_send?
+      title
+    end
+
+  private
+
+    def is_modern_lanuage_course?(subjects)
+      subjects[0] == SecondarySubject.modern_languages
```

Comment:
will that be an issue? always first....

Reply:
I would worry that, similar to below, `subjects` is all types of "subjects" ... maybe this should be:

```suggestion
      primary_subjects.include? SecondarySubject.modern_languages
```